docs/upgrading/upgrading_to_v1.md

```diff
@@ -0,0 +1,123 @@
+---
+id: upgrading-to-v1
+title: Upgrading to v1
+---
+
+This page summarizes the breaking changes between Crawlee for Python v0.6 and v1.0.
+
+## Storage clients
+
+In v1.0, we are introducing a new storage clients system. We have completely reworked their interface,
+making it much simpler to write your own storage clients. This allows you to easily store your request queues,
+key-value stores, and datasets in various destinations.
+
+### New storage clients
+
+Previously, the `MemoryStorageClient` handled both in-memory storage and file system persistence, depending
+on configuration. In v1.0, we've split this into two dedicated classes:
+
+- `MemoryStorageClient` - stores all data in memory only.
+- `FileSystemStorageClient` - persists data on the file system, with in-memory caching for improved performance.
+
+For details about the new interface, see the `BaseStorageClient` documentation. You can also check out
+the [Storage clients guide](https://crawlee.dev/python/docs/guides/) for more information on available
+storage clients and instructions on writing your own.
+
+### Memory storage client
+
+Before:
+
+```python
+from crawlee.configuration import Configuration
+from crawlee.storage_clients import MemoryStorageClient
+
+configuration = Configuration(persist_storage=False)
+storage_client = MemoryStorageClient.from_config(configuration)
+```
+
+Now:
+
+```python
+from crawlee.storage_clients import MemoryStorageClient
+
+storage_client = MemoryStorageClient()
+```
+
+### File-system storage client
+
+Before:
+
+```python
+from crawlee.configuration import Configuration
+from crawlee.storage_clients import MemoryStorageClient
+
+configuration = Configuration(persist_storage=True)
+storage_client = MemoryStorageClient.from_config(configuration)
+```
+
+Now:
+
+```python
+from crawlee.storage_clients import FileSystemStorageClient
+
+storage_client = FileSystemStorageClient()
+```
+
+The way you register storage clients remains the same:
+
+```python
+from crawlee import service_locator
+from crawlee.crawlers import ParselCrawler
+from crawlee.storage_clients import MemoryStorageClient
+
+storage_client = MemoryStorageClient()
+
+# Either via the service locator:
+service_locator.set_storage_client(storage_client)
+
+# Or provide it directly to the crawler:
+crawler = ParselCrawler(storage_client=storage_client)
+```
+
+### Breaking changes
```

Comment:
It'd be fair to mention that when you call for example `StorageClient.open_dataset_client()` or `DatasetClient.open()` twice with the same arguments, you may get different instances of `DatasetClient`. It's not a frequent use case by any means, but let's strive to maintain order.

Reply:
Ok, I will address it.